kbase-extension/static/kbase/js/widgets/function_input/kbaseGrowthParamsPlotInput.js

```diff
@@ -238,46 +236,61 @@ define(['jquery',
+                                var matrixRef = self.growthParams.matrix_id;
+                                if( !matrixRef ) return;
+                    
+                                var query = [];
+                                query.push( {ref: matrixRef, included: ["metadata/column_metadata"]} );
 
+                                self.state = self.STATE_FETCHING;                    
+                                self.ws.get_object_subset(query,
+                                    function(result) {
+                                        self.columnMetadata = result[0].data.metadata.column_metadata;
+                                        var conditions = self.getConditions(self.columnMetadata);
 
-                        self.state = self.STATE_FETCHING;                    
-                        self.ws.get_object_subset(query,
-                            function(result) {
-                                self.columnMetadata = result[0].data.metadata.column_metadata;
-                                var conditions = {};
-                                for(var i in self.columnMetadata){
-                                    var cm = self.columnMetadata[i];
-                                    for(var j in cm){
-                                        var propValue = cm[j];
-                                        if(propValue.entity != 'Condition') continue;
-                                        conditions[propValue.property_name] = propValue.property_name;
-                                    }
-                                }
+                                        self.conditionParams = [];
+                                        for(var conditionParam in conditions){
+                                            self.conditionParams.push({id: conditionParam, text: conditionParam});
+                                        }
 
-                                self.conditionParams = [];
-                                for(var i in conditions){
-                                    var conditionParam = conditions[i];
-                                    self.conditionParams.push({id: conditionParam, text: conditionParam});
-                                }
-
-                                self.state = self.STATE_READY;
-                                if(doneCallback) { doneCallback(self.conditionParams); }
+                                        self.state = self.STATE_READY;
+                                        if(doneCallback) { doneCallback(self.conditionParams); }
+                                    },
+                                    function(error) {
+                                        console.error(error);
+                                    }
+                                );                        
                             },
-                            function(error) {
+                            function(eror){
```

Comment:
This'll cause an error if an error gets thrown - the 'eror' name should be 'error' for the console message.